cpp/include/rapidsmpf/memory/pinned_memory_resource.hpp

```diff
@@ -32,15 +33,22 @@ namespace rapidsmpf {
  * @brief Checks if the PinnedMemoryResource is supported for the current CUDA version.
  *
  * RapidsMPF requires CUDA 12.6 or newer to support pinned memory resources.
- *
- * @note The driver version check is cached and only performed once.
  */
 inline bool is_pinned_memory_resources_supported() {
 #if RAPIDSMPF_CUDA_VERSION_AT_LEAST(RAPIDSMPF_PINNED_MEM_RES_MIN_CUDA_VERSION)
     static const bool supported = [] {
-        int driver_version = 0;
-        RAPIDSMPF_CUDA_TRY(cudaDriverGetVersion(&driver_version));
-        return driver_version >= RAPIDSMPF_PINNED_MEM_RES_MIN_CUDA_VERSION;
+        // check if the device supports async memory pools
+        if (!rmm::detail::runtime_async_alloc::is_supported()) {
+            return false;
+        }
+
+        int cuda_driver_version{};
+        auto driver_result = cudaDriverGetVersion(&cuda_driver_version);
+        int cuda_runtime_version{};
+        auto runtime_result = cudaRuntimeGetVersion(&cuda_runtime_version);
+        return driver_result == cudaSuccess && runtime_result == cudaSuccess
+               && cuda_driver_version >= RAPIDSMPF_PINNED_MEM_RES_MIN_CUDA_VERSION
+               && cuda_runtime_version >= RAPIDSMPF_PINNED_MEM_RES_MIN_CUDA_VERSION;
```

Comment:
I suggest you move this to a separate PR that only address the _is-supported_ issue and tests the different access types.